src/test/java/com/haxwell/apps/quizki/services/QuestionServiceImplTest.java

```diff
@@ -274,5 +335,71 @@ public void getQuestionByIdTest() {
 		assertThat(outputDTO.getId(), equalTo(1l));
 		
 	}
+	
+	@Test
+	public void getQuestionsTest() {
+		
+		
+		/*
+		 * Stated requirements
+		 * URL: /api/question?page=1&size=10
+		 * URL: /api/question?page=1 (size defaults to 10)
+		 * URL: /api/question?size=10 (page defaults to 1)
+		 * URL: /api/question (page and size are default)
+		 * 
+		 * parameters are passed to getQuestions() as received by the controller with defaults
+		 */
+		
+		int n = 17;		//TODO: refactor the magic numbers in the assertions based on this value
+		questions = getQuestions(n);
+		
+		when(questionRepo.count()).thenReturn((long)n);
+		
+		when(questionRepo.findAll(any(Pageable.class))).thenAnswer(new Answer<Page<Question>>() {
```

Comment:
You should be able to return `List<Question>` from here.
```
when(questionRepo.findAll(any(Pageable.class))).thenReturn(questions);
```
That should work.. you'd have to rewrite the repo method to return a `List<Question>` but it should work.
 Methinks what you have here is more low-level than necessary.

Reply:
Sure, a List would be easier to work with. I wind up converting it anyway.